x-pack/plugins/fleet/server/routes/package_policy/utils/index.ts

```diff
@@ -53,3 +65,47 @@ export function canUseMultipleAgentPolicies() {
       : 'Reusable integration policies are not supported',
   };
 }
+
+/**
+ * If an agentless agent policy is associated with the package policy,
+ * it will rename the agentless agent policy of a package policy to keep it in sync with the package policy name.
+ */
+export async function renameAgentlessAgentPolicy(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  packagePolicy: PackagePolicy,
+  name: string
+) {
+  const cloudSetup = appContextService.getCloud();
+  // If cloud and agentless is enabled, we need to rename the agent policy
+  // tech debt: update this condition when Serverless uses the Agentless API
+  // https://github.com/elastic/security-team/issues/9781
+  if (cloudSetup?.isCloudEnabled && appContextService.getExperimentalFeatures().agentless) {
```

Comment:
This series of checks are duplicated in [createAgentlessAgent in agent_policy service](https://github.com/elastic/kibana/pull/187488#top)

I've created a tech debt here: 
- https://github.com/orgs/elastic/projects/705/views/91?sliceBy%5Bvalue%5D=Agentless+-+API+-+ESS&pane=issue&itemId=70638033

Reply:
Thanks!